addons/account/controllers/portal.py

```diff
@@ -60,6 +60,9 @@ def _get_account_searchbar_filters(self):
 
     @http.route(['/my/invoices', '/my/invoices/page/<int:page>'], type='http', auth="user", website=True)
     def portal_my_invoices(self, page=1, date_begin=None, date_end=None, sortby=None, filterby=None, **kw):
+        visibility = self._check_page_visibility("account.portal_my_home_invoice")
+        if not visibility:
+            return request.not_found()
```

Comment:
```suggestion
        if not self._check_page_visibility("account.portal_my_home_invoice"):
            return request.not_found()
```

Follow the same structure for both pr.